datadog_checks_base/datadog_checks/base/utils/agent/memory.py

```diff
@@ -121,9 +129,11 @@ def write_pretty_top(path, snapshot, unit_formatter, key_type, limit, cumulative
         amount, unit = unit_formatter(total)
         f.write('Total allocated size: {} {}\n'.format(amount, unit))
 
+    metrics.append(MemoryProfileMetric('check_run_total', total))
```

Comment:
maybe `check_run_alloc`? I feel like total might mislead to think it's the entire memory being used by the check, where that wouldn't be entirely accurate.